pkg/engine/engine.go

```diff
@@ -150,51 +151,61 @@ func (e Engine) renderWithReferences(tpls, referenceTpls map[string]renderable)
 	// higher-level (in file system) templates over deeply nested templates.
 	keys := sortTemplates(tpls)
 
-	for _, fname := range keys {
-		r := tpls[fname]
-		if _, err := t.New(fname).Parse(r.tpl); err != nil {
-			return map[string]string{}, errors.Wrapf(err, "parse error in %q", fname)
+	for _, filename := range keys {
+		r := tpls[filename]
+		if _, err := t.New(filename).Parse(r.tpl); err != nil {
+			return map[string]string{}, parseTemplateError(err)
 		}
 	}
 
 	// Adding the reference templates to the template context
 	// so they can be referenced in the tpl function
-	for fname, r := range referenceTpls {
-		if t.Lookup(fname) == nil {
-			if _, err := t.New(fname).Parse(r.tpl); err != nil {
-				return map[string]string{}, errors.Wrapf(err, "parse error in %q", fname)
+	for filename, r := range referenceTpls {
+		if t.Lookup(filename) == nil {
+			if _, err := t.New(filename).Parse(r.tpl); err != nil {
+				return map[string]string{}, parseTemplateError(err)
 			}
 		}
 	}
 
 	rendered = make(map[string]string, len(keys))
-	for _, file := range keys {
+	for _, filename := range keys {
 		// Don't render partials. We don't care out the direct output of partials.
 		// They are only included from other templates.
-		if strings.HasPrefix(path.Base(file), "_") {
+		if strings.HasPrefix(path.Base(filename), "_") {
 			continue
 		}
 		// At render time, add information about the template that is being rendered.
-		vals := tpls[file].vals
-		vals["Template"] = chartutil.Values{"Name": file, "BasePath": tpls[file].basePath}
+		vals := tpls[filename].vals
+		vals["Template"] = chartutil.Values{"Name": filename, "BasePath": tpls[filename].basePath}
 		var buf strings.Builder
-		if err := t.ExecuteTemplate(&buf, file, vals); err != nil {
-			return map[string]string{}, errors.Wrapf(err, "render error in %q", file)
```

Comment:
If there's an error, the filename will be included in `template`'s error message, which is parsed out [here](https://github.com/helm/helm/pull/5572/files/33b1ede570aec8266eda547c0d8798c9e02f1064#diff-6bb0b5c36296ab58781ffd3f44d868f1R203). But if we do happen to run into the array length error you mentioned, we'll be returning the original error message. Do we want to include the filename regardless? It will show up twice in the message, but will be less likely to fail

Reply:
Nah, the current output is great. I just need more coffee. :)